nasa/harmony
# DAS-2157 - Update HARMONY_REGRIDDER_SERVICE to ghcr.io.

## Jira Issue ID

DAS-2157

## Description

This PR switches Harmony over to using [a Docker image hosted in the GitHub Container Registry](https://github.com/nasa/harmony-regridding-service/pkgs/container/harmony-regridding-service) for the Harmony Regridding Service (the latest repository to migrate to open-source).  There are no changes to the names of any of the environment variables, just a couple of the values, so this is a pretty lightweight PR.

I'll make similar changes in the harmony-ci-cd repository and submit a PR for those today.

## Local Test Steps

* Retrieve the new Docker image from ghcr.io: `docker pull ghcr.io/nasa/harmony-regridding-service:latest`.
* Pull this branch of the Harmony repository locally.
* Build a new version of the Harmony images: `npm run build-all` and then `npm run build` (for the actual `harmony` Docker image).
* Bootstrap Harmony: `./bin/bootstrap-harmony`.
* Create a new Python environment (conda or pyenv) and run `pip install harmony-py`.
* Run the snippet below. It should trigger a request to Harmony.
* In the workflow-ui, confirm that the second step for the request goes to `nasa/harmony-regridding-service`.
* Feel free to download the result and check it's a decent NetCDF-4 file, but the main test here is that the new Docker image is used and the request gets routed to the right container.

```
from harmony import Collection, Environment, Client, Request


harmony_host_url = 'http://localhost:3000'
harmony_client = Client(env=Environment.LOCAL)

merra_request = Request(
    collection=Collection(id='C1245662776-EEDTEST'),
    granule_id='G1245662793-EEDTEST',
    interpolation='Elliptical Weighted Averaging',
    scale_size=(2.5, 2.5),
    scale_extent=(-180, -90, 180, 90),
    crs='+proj=latlong +datum=WGS84 +no_defs',
    height=72,
    width=144
)                       
merra_job_id = harmony_client.submit(merra_request)
harmony_client.wait_for_processing(merra_job_id, show_progress=True)
```

## PR Acceptance Checklist
* [x] Tests added/updated (if needed) and passing (they are unchanged, but passing)
